app/prepare_proposal.go

```diff
@@ -33,19 +33,27 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 	}
 
 	// in this step we are processing any MsgWirePayForBlob transactions into
-	// MsgPayForBlob and their respective messages. The malleatedTxs contain the
+	// MsgPayForBlob and their respective blobPointers. The malleatedTxs contain the
 	// the new sdk.Msg with the original tx's metadata (sequence number, gas
 	// price etc).
-	processedTxs, messages, err := malleateTxs(app.txConfig, squareSize, parsedTxs, req.BlockData.Evidence)
+	processedTxs, blobPointers, err := malleateTxs(app.txConfig, squareSize, parsedTxs, req.BlockData.Evidence)
 	if err != nil {
 		panic(err)
 	}
 
+	blobs := make([]core.Blob, len(blobPointers))
+	for i, blob := range blobPointers {
+		blobs[i] = core.Blob{
+			NamespaceId: blob.GetNamespaceId(),
+			Data:        blob.GetData(),
+		}
+	}
```

Comment:
[optional]

see other comment on how we might be able to remove this extra conversion. if not, do you think we would get any benefit out of dereferencing instead of creating an entire new struct?

it doesn't really matter, as we are going to change this soon. pls move along if its not a quick change :pray:

Reply:
Resolved in https://github.com/celestiaorg/celestia-app/pull/1023/commits/3543e6d9c92760542b0396ac6a918c69b0e4be22